service/lib/dinstaller/storage/iscsi/manager.rb

```diff
@@ -0,0 +1,225 @@
+# frozen_string_literal: true
+
+# Copyright (c) [2023] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "yast"
+require "dinstaller/storage/iscsi/node"
+require "dinstaller/storage/iscsi/initiator"
+
+Yast.import "IscsiClientLib"
+
+module DInstaller
+  module Storage
+    module ISCSI
+      # Manager for iSCSI
+      class Manager
+        # iSCSI initiator
+        #
+        # @return [Initiator]
+        attr_reader :initiator
+
+        # Discovered iSCSI nodes
+        #
+        # @return [Array<Node>]
+        attr_reader :nodes
+
+        # Constructor
+        #
+        # @param logger [Logger, nil]
+        def initialize(logger: nil)
+          @logger = logger || ::Logger.new($stdout)
+          @initiator = ISCSI::Initiator.new
+
+          @on_probe_callbacks = []
+        end
+
+        # Performs actions for activating iSCSI
+        def activate
+          return if @activated
```

Comment:
I'm fine with this. But actually I think running activation twice would be safe.

All current operations are safe.

The original goal of having an `@activated` flag (well, it was called `@probed` back then) was to ensure other operations were never performed before a probing/activation. Not preventing the activation/probing to be run twice.

For example, you should never try to discover is you have not activated before (because maybe there is not a valid initiator name).